### PR TITLE
feat(compaction): use trivial move to convert no-overlapping-level

### PR DESCRIPTION
**Pull request by Little-Wallace**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

https://github.com/risingwavelabs/risingwave/pull/9794 has been reverted since it caused bug mentioned by https://github.com/risingwavelabs/risingwave/issues/10240.
This PR will implement the same feature as https://github.com/risingwavelabs/risingwave/pull/9794 but it will not change the logic of `apply_verion_delta`. Instead of, it will trigger another compact task to get the same effort.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [x] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

## Documentation

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by Little-Wallace** [comment]
image

We can see that after split, group-5 could avoid compact a sorted sub-level in compactor. So the write throughput for level0 down to zero.

But there is some corner case that group-3 still compact some sub-level before split.  I would give another PR and solve it in compactor.

**Comment by Little-Wallace** [comment]
I changed checkpoint frequency to 4. And this PR almost convert all øverlapping level by trivial move.
It is because that when the checkpoint frequency is too large, we would keep more data in shared-buffer until it triggers a `spill flush`. This operation would make sstables of level0 overlap with each other

image

**Review by Li0k — APPROVED** [review]
Rest LGTM

File: src/meta/src/hummock/compaction/level_selector.rs [hunk 1]
```diff
@@ -275,7 +275,8 @@ impl DynamicLevelSelectorCore {
         }
 
         // sort reverse to pick the largest one.
-        ctx.score_levels.sort_by(|a, b| b.0.cmp(&a.0));
```
[the remark this answers is not in the document]
Comment (Little-Wallace) [reply src/meta/src/hummock/compaction/level_selector.rs:278]: Just to pass some test

[the remark this answers is not in the document]
Comment (Little-Wallace) [reply src/meta/src/hummock/compaction/level_selector.rs:278]: I think it make no difference

File: src/meta/src/hummock/manager/mod.rs [hunk 2]
```diff
@@ -908,50 +909,31 @@ where
...
-                    };
-                    format!(
-                        "L0->L{} {}",
-                        compact_task.input_ssts.last().unwrap().level_idx,
-                        is_trivial_move
-                    )
-                };
-
-                let level_count = compact_task.input_ssts.len();
+            let level_type_label = format!(
```
[the remark this answers is not in the document]
Comment (Little-Wallace) [reply src/meta/src/hummock/manager/mod.rs:912]: Because it is not necessary to collect `compact_task_size` for trivial-move task

**Pull request merged**